src/context.cpp

```diff
@@ -596,61 +699,172 @@ namespace librealsense
             if ( profiles.size() > 0 )
             {
                 _dev->open( realdds_profiles );
+
             }
 
             software_sensor::open( profiles );
         }
 
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                std::shared_ptr< stream_profile_interface > prof, std::string name )
+        {
+            rs2_software_video_frame rs2_frame = {};
+
+            // prof parameter holds the real data, rs2_software_video_frame forces us to hold a pointer to it.
+            // Because we use syncer, not calling on_video_frame in the lifetime of this function, we need a shared_ptr
+            // that the syncer will hold till using the frame.
+            std::shared_ptr< rs2_stream_profile > prof_holder = std::make_shared< rs2_stream_profile >();
```

Comment:
If we need a holder to hold information about the frame (whether an extra smart ptr for lifetime or otherwise), how about:
* define a "synced frame" object that has just the relevant information needed by the syncer (i.e., frame-id)
* contains a pointer to the actual frame (as a void*), possibly with a deleter
* if needed, contains any additional pointers (like this holder)

So this class doesn't have to be a template. You deal with your own frame objects and only the callbacks need to know how to interpret the frame (and metadata).

Reply:
I don't see how to avoid the template using the current `rs2_software_video_frame` and `rs2_motion_video_frame` API structs. The syncer still needs to call `on_video_frame`/`on_motion_frame` with the correct type.

Changes to the API should be done carefully and in a different PR.